### ISSUE TITLE
Differ between error states in liftajax.js

### ISSUE THREAD

**Issue by dhobi** [lift/framework#1320]
LiftRules.ajaxDefaultFailure does currently not distinguish between different types of errors (i.e. non-redirecting status codes). Therefore an ajax call resulting in a 500 has the same effect as an ajax call timing out.

Please see:
https://groups.google.com/forum/?fromgroups=#!topic/liftweb/Je7yX6R76LQ

**Comment by Shadowfiend** [comment]
This could be a simple matter of updating the [`ajaxOnFailure` callback in `lift.js`](https://github.com/lift/framework/blob/master/web/webkit/src/main/resources/toserve/lift.js#L52) to be passed some additional parameters [when it is invoked](https://github.com/lift/framework/blob/master/web/webkit/src/main/resources/toserve/lift.js#L256). However, the question is *how* those parameters are passed. In particular, jQuery normalizes some of the data the browser provides to build things like an aborted status vs a timeout status, etc. The `liftVanilla` implementation will have to be able to do something similar before we can fully support this.

**Comment by dhobi** [comment]
Back then we just differed between "timeout" (string comparison) and "the rest" (http status code):
https://github.com/dhobi/liftajax/blob/master/src/main/scala/code/util/MyScriptRenderer.scala#L147

Don't know if this much of a help, just thought I'll share our somewhat hacky solution.

**Comment by Shadowfiend** [comment]
Aye, I skimmed that. I think there's value to going all the way here and exposing status code and cause-as-string somehow. Again, the core issue will be the `liftVanilla` implementation that doesn't rely on jquery---but it could be that that will also be easy. I'm not sure, just wanted to give a brief “this is how this could go” blurb and didn't have time to investigate deeply how hard it will be to differentiate those possibilities.
